providers/azure/resources/compute.go

```diff
@@ -79,6 +96,48 @@ func (a *mqlAzureSubscriptionComputeService) vms() ([]interface{}, error) {
 	return res, nil
 }
 
+func (a *mqlAzureSubscriptionComputeServiceVm) state() (string, error) {
```

Comment:
for getting the state, we need to call `InstanceView` for each machine. It does not look like this information is present from the bulk retrieval API